packages/salesforcedx-vscode-core/src/metadataSupport/metadataHoverProvider.ts

```diff
@@ -0,0 +1,255 @@
+    const fieldInfo = this.extractFieldInfo(document, position);
+    if (fieldInfo) {
+      const fieldDocumentation = await this.documentationService.getFieldDocumentation(
+        fieldInfo.metadataType,
+        fieldInfo.fieldName
+      );
+
+      if (fieldDocumentation) {
+        const markdownContent = new vscode.MarkdownString();
+        markdownContent.appendCodeblock(`${fieldInfo.metadataType}.${fieldInfo.fieldName}`, 'xml');
+        markdownContent.appendMarkdown(fieldDocumentation.description);
+
+        if (fieldDocumentation.type) {
+          markdownContent.appendMarkdown(`\n\n**Type:** \`${fieldDocumentation.type}\``);
+        }
+
+        if (fieldDocumentation.required !== undefined) {
+          markdownContent.appendMarkdown(`\n**Required:** ${fieldDocumentation.required ? 'Yes' : 'No'}`);
+        }
+
+        if (fieldDocumentation.validValues && fieldDocumentation.validValues.length > 0) {
+          markdownContent.appendMarkdown('\n\n**Valid Values:**\n');
+          fieldDocumentation.validValues.forEach(value => {
+            markdownContent.appendMarkdown(`- \`${value}\`\n`);
+          });
+        }
+
+        return new vscode.Hover(markdownContent, wordRange);
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Check if the document is likely a Salesforce metadata file
+   */
+  private isMetadataFile(document: vscode.TextDocument): boolean {
+    // Since we're now registered for all XML files, check if this is a Salesforce metadata file
+    // by looking for the Salesforce metadata namespace or common metadata file patterns
+    const fileName = path.basename(document.fileName);
+    const documentText = document.getText();
+
+    // Check for Salesforce metadata namespace
+    if (documentText.includes('http://soap.sforce.com/2006/04/metadata')) {
+      return true;
+    }
+
+    // Check for common Salesforce metadata file naming patterns
+    return (
+      fileName.endsWith('-meta.xml') ||
+      fileName.includes('.object') ||
+      fileName.includes('.flow') ||
+      fileName.includes('.layout') ||
+      fileName.includes('.profile') ||
+      fileName.includes('.permissionset') ||
+      fileName.includes('.profilePasswordPolicy') ||
+      fileName.includes('.app') ||
+      fileName.includes('.tab') ||
+      fileName.includes('.trigger') ||
+      fileName.includes('.cls') ||
+      fileName.includes('.component') ||
+      fileName.includes('.page') ||
+      fileName.includes('.email') ||
+      fileName.includes('.report') ||
+      fileName.includes('.dashboard') ||
+      fileName.includes('.resource') ||
+      fileName.includes('.workflow') ||
+      fileName.includes('.validationRule') ||
+      fileName.includes('.customField')
+    );
+  }
+
+  /**
+   * Extract metadata type or field from the current line and cursor position
+   */
+  private extractMetadataType(lineText: string, word: string, cursorPosition: number): string | null {
```

Comment:
here and elsewhere: if it doesn't use `this` it shouldn't be a method.  put a fn in module scope.  Then if you want you can write unit tests for it without having to create a class, easily refactor, etc.